plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -3150,7 +3150,9 @@ protected Answer execute(StopCommand cmd) {
                     vmMo.setCustomFieldValue(CustomFieldConstants.CLOUD_VM_INTERNAL_NAME, cmd.getVmName());
 
                     if (getVmPowerState(vmMo) != PowerState.PowerOff) {
-                        if (vmMo.safePowerOff(_shutdownWaitMs)) {
+                        if (cmd.isForceStop() && vmMo.powerOff()) {
+                            return new StopAnswer(cmd, "Force stopped VM " + cmd.getVmName() + " Succeed", true);
```

Comment:
Better to keep the response (StopAnswer message) similar for all the HVs. The same should be done for XS and KVM as well.